### ISSUE TITLE
Manager should use primaries of Object attributes when saving

### ISSUE THREAD

**Issue by pierrelemee** [shabbyrobe/amiss#23]
When I try to save an entity having a field related to another table (using the `@has.one.of` clause) I get an Exception because my object is not a valid type.

`Manager` should check if the value is an object, and if so if the underlying class has a meta in order to use its primary value.

This in order to avoid painful temporary settings, like:

``` PHP
$temp = $entity->getUser();
$entity->setUser($temp->getId());
$manager->save($entity);
$entity->setUser($temp);
```

@shabbyrobe do you think this kind of request makes sense ?

**Comment by shabbyrobe** [comment]
Is this related to PR #24?

**Comment by pierrelemee** [comment]
Yes it is! Sorry I'm not used to Github's issues and pull-requests.

  *Pierre LEMÉE*
*Développeur*
http://www.pierrelemee.fr
(+33) 06.78.58.57.40

2016-08-26 10:02 GMT+02:00 Blake Williams <notifications@github.com>:

**Comment by shabbyrobe** [comment]
No worries, @pierrelemee, I worked it out :)

Did you see my reply in the #24 thread?

**Comment by pierrelemee** [comment]
Oh yes sorry! Let me answer there ;)
